gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchTest.java

```diff
@@ -0,0 +1,224 @@
+    Capture<RpcBatch.Callback<StorageObject>> callback = Capture.newInstance();
+    batchMock.addPatch(EasyMock.eq(BLOB_INFO.toPb()), EasyMock.capture(callback),
+        EasyMock.eq(ImmutableMap.<StorageRpc.Option, Object>of()));
+    EasyMock.replay(batchMock);
+    StorageBatchResult<Blob> batchResult = dnsBatch.update(BLOB_INFO);
+    assertNotNull(callback.getValue());
+    try {
+      batchResult.get();
+      fail("No result available yet.");
+    } catch (IllegalStateException ex) {
+      // expected
+    }
+    // testing error here, success is tested with options
+    RpcBatch.Callback<StorageObject> capturedCallback = callback.getValue();
+    capturedCallback.onFailure(GOOGLE_JSON_ERROR);
+    try {
+      batchResult.get();
+      fail("Should throw a StorageExcetion on error.");
+    } catch (StorageException ex) {
+      // expected
+    }
+  }
+
+  @Test
+  public void testUpdateWithOptions() {
+    EasyMock.reset(storage, batchMock, optionsMock);
+    EasyMock.expect(storage.options()).andReturn(optionsMock).times(2);
+    EasyMock.expect(optionsMock.service()).andReturn(storage);
+    Capture<RpcBatch.Callback<StorageObject>> callback = Capture.newInstance();
+    Capture<Map<StorageRpc.Option, Object>> capturedOptions = Capture.newInstance();
+    batchMock.addPatch(EasyMock.eq(BLOB_INFO_COMPLETE.toPb()), EasyMock.capture(callback),
+        EasyMock.capture(capturedOptions));
+    EasyMock.replay(batchMock, storage, optionsMock);
+    StorageBatchResult<Blob> batchResult = dnsBatch.update(BLOB_INFO_COMPLETE, BLOB_TARGET_OPTIONS);
+    assertNotNull(callback.getValue());
+    assertEquals(2, capturedOptions.getValue().size());
+    assertEquals(42L, capturedOptions.getValue().get(BLOB_TARGET_OPTIONS[0].rpcOption()));
+    assertEquals(42L, capturedOptions.getValue().get(BLOB_TARGET_OPTIONS[1].rpcOption()));
+    RpcBatch.Callback<StorageObject> capturedCallback = callback.getValue();
+    capturedCallback.onSuccess(BLOB_INFO.toPb());
+    assertEquals(new Blob(storage, new Blob.BuilderImpl(BLOB_INFO)), batchResult.get());
+  }
+
+  @Test
+  public void testGet() {
+    EasyMock.reset(batchMock);
+    Capture<RpcBatch.Callback<StorageObject>> callback = Capture.newInstance();
+    batchMock.addGet(EasyMock.eq(BLOB_INFO.toPb()), EasyMock.capture(callback),
+        EasyMock.eq(ImmutableMap.<StorageRpc.Option, Object>of()));
+    EasyMock.replay(batchMock);
+    StorageBatchResult<Blob> batchResult = dnsBatch.get(BLOB_ID.bucket(), BLOB_ID.name());
+    assertNotNull(callback.getValue());
+    try {
+      batchResult.get();
+      fail("No result available yet.");
+    } catch (IllegalStateException ex) {
+      // expected
+    }
+    // testing error here, success is tested with options
+    RpcBatch.Callback<StorageObject> capturedCallback = callback.getValue();
+    capturedCallback.onFailure(GOOGLE_JSON_ERROR);
+    try {
+      batchResult.get();
+      fail("Should throw a StorageExcetion on error.");
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2469779062/issues/source?bid=3123103&fileBranchId=3247265#l198)